docs/en/sql-reference/functions/random-functions.md

```diff
@@ -11,79 +11,211 @@ elimination](../../sql-reference/functions/index.md#common-subexpression-elimina
 function return different random values.
 
 Related content
+
 - Blog: [Generating random data in ClickHouse](https://clickhouse.com/blog/generating-random-test-distribution-data-for-clickhouse)
 
 :::note
 The random numbers are generated by non-cryptographic algorithms.
 :::
 
-## rand, rand32
+## rand
+
+Returns a random UInt32 number.
+
+### Syntax
+
+```sql
+rand()
+```
+
+### Parameters
+
+None.
 
-Returns a random UInt32 number, evenly distributed across the range of all possible UInt32 numbers.
+### Output
+
+Returns a number of type UInt32.
+
+### Implementation details
 
 Uses a linear congruential generator.
 
+### Example
+
+```sql
+SELECT rand();
+```
+
+```response
+1569354847
+```
+
+**Note:** The actual output will be a random number, not the specific number shown in the example.
+
+## rand32
+
+Returns a random 32-bit unsigned integer (UInt32) number.
+
+### Syntax
+
+```sql
+rand32()
+```
+
+### Parameters
+
+None.
+
+### Output
+
+Returns a number of type UInt32, evenly distributed across the range of all possible UInt32 values.
+
+### Example
+
+```sql
+SELECT rand32();
+```
+
+```response
+2754546224
+```
+
+**Note:** The actual output will be a random number, not the specific number shown in the example.
+
 ## rand64
 
-Returns a random UInt64 number, evenly distributed across the range of all possible UInt64 numbers.
+Returns a random 64-bit unsigned integer (UInt64) number.
 
-Uses a linear congruential generator.
+### Syntax
+
+```sql
+rand64()
+```
+
+### Parameters
+
+None.
+
+### Output
+
+Returns a number of type UInt64, evenly distributed across the range of all possible UInt64 values.
+
+### Implementation details
+
+The `rand64` function uses a linear congruential generator, which means that while it appears random, it's not truly random and can be predictable if the initial state is known. For scenarios where true randomness is crucial, consider using alternative methods like system-level calls or integrating with external libraries.
```

Comment:
It uses a truncated linear congruential generator with an initial state obtained from the system.